src/HTSPDemuxer.cpp

```diff
@@ -454,82 +458,80 @@ void CHTSPDemuxer::ParseSubscriptionStart ( htsmsg_t *m )
       }
 
       /* Language */
-      if (m_streams.stream[count].iCodecType == XBMC_CODEC_TYPE_SUBTITLE ||
-          m_streams.stream[count].iCodecType == XBMC_CODEC_TYPE_AUDIO)
+      if (stream.iCodecType == XBMC_CODEC_TYPE_SUBTITLE ||
+          stream.iCodecType == XBMC_CODEC_TYPE_AUDIO)
       {
         const char *language;
         
         if ((language = htsmsg_get_str(&f->hmf_msg, "language")) != NULL)
-          strncpy(m_streams.stream[count].strLanguage, language, sizeof(m_streams.stream[count].strLanguage) - 1);
+          strncpy(stream.strLanguage, language, sizeof(stream.strLanguage) - 1);
       }
 
       /* Audio data */
-      if (m_streams.stream[count].iCodecType == XBMC_CODEC_TYPE_AUDIO)
+      if (stream.iCodecType == XBMC_CODEC_TYPE_AUDIO)
       {
-        m_streams.stream[count].iChannels
-          = htsmsg_get_u32_or_default(&f->hmf_msg, "channels", 2);
-        m_streams.stream[count].iSampleRate
-          = htsmsg_get_u32_or_default(&f->hmf_msg, "rate", 48000);
+        stream.iChannels = htsmsg_get_u32_or_default(&f->hmf_msg, "channels", 2);
+        stream.iSampleRate = htsmsg_get_u32_or_default(&f->hmf_msg, "rate", 48000);
       }
 
       /* Video */
-      if (m_streams.stream[count].iCodecType == XBMC_CODEC_TYPE_VIDEO)
+      if (stream.iCodecType == XBMC_CODEC_TYPE_VIDEO)
       {
-        m_streams.stream[count].iWidth  = htsmsg_get_u32_or_default(&f->hmf_msg, "width", 0);
-        m_streams.stream[count].iHeight = htsmsg_get_u32_or_default(&f->hmf_msg, "height", 0);
+        stream.iWidth  = htsmsg_get_u32_or_default(&f->hmf_msg, "width", 0);
+        stream.iHeight = htsmsg_get_u32_or_default(&f->hmf_msg, "height", 0);
         
         /* Ignore this message if the stream details haven't been determined 
            yet, a new message will be sent once they have. This is fixed in 
            some versions of tvheadend and is here for backward compatibility. */
-        if (m_streams.stream[count].iWidth == 0 || m_streams.stream[count].iHeight == 0)
+        if (stream.iWidth == 0 || stream.iHeight == 0)
         {
           Logger::Log(LogLevel::LEVEL_DEBUG, "Ignoring subscriptionStart, stream details missing");
           return;
         }
         
         /* Setting aspect ratio to zero will cause XBMC to handle changes in it */
-        m_streams.stream[count].fAspect = 0.0f;
+        stream.fAspect = 0.0f;
         
         if ((u32 = htsmsg_get_u32_or_default(&f->hmf_msg, "duration", 0)) > 0)
         {
-          m_streams.stream[count].iFPSScale = u32;
-          m_streams.stream[count].iFPSRate  = DVD_TIME_BASE;
+          stream.iFPSScale = u32;
+          stream.iFPSRate  = DVD_TIME_BASE;
         }
       }
         
-      Logger::Log(LogLevel::LEVEL_DEBUG, "  id: %d, type %s, codec: %u", idx, type, m_streams.stream[count].iCodecId);
+      Logger::Log(LogLevel::LEVEL_DEBUG, "  id: %d, type %s, codec: %u", idx, type, stream.iCodecId);
       count++;
```

Comment:
The variable <code>count</code> can be removed. You're not doing anything useful with it anymore. ;-)